src/pretalx/agenda/views/schedule.py

```diff
@@ -114,7 +114,10 @@ def get(self, request, *args, **kwargs):
 
         exporter.schedule = self.schedule
         if "-my" in exporter.identifier and self.request.user.id is None:
-            return HttpResponseRedirect(self.request.event.urls.login)
+            if request.GET.get('talks'):
```

Comment:
Here is the original code:

```py
if "-my" in exporter.identifier and self.request.user.id is None:
    return HttpResponseRedirect(self.request.event.urls.login)
favs_talks = SubmissionFavourite.objects.filter(user=self.request.user.id)
```

The author wanted to prevent `SubmissionFavourite.objects.filter()` from receiving `None`, so there is the check `self.request.user.id is None`, to force user logging-in.

But you insert your code in this `if` flow control and it nullify the guard. After your code, `return HttpResponseRedirect` doesn't run and `None` strips to `SubmissionFavourite.objects.filter`.

Reply:
Hi @hongquan, this view is using for both talk and video component:
1. From talk front end API call, 'talks' won't be put as request param, so if User not login, they will be forced to login page.
2. From video API call, 'talks' is input as request params and from video, we not expecting User will be redirect to Talk component to login.

Idea for this changes is reused the current API for Video schedule export, not impact flow at talk component

Reply:
@lcduong Could you add your explanation as comment above that line?